src/game/Game.java

```diff
@@ -107,99 +104,120 @@ public int getPlayerTurn() {
+			RegularMove moveToMake = new RegularMove(boardState, start, dest);
+			this.currentBoard = moveToMake.execute();
 			this.moves.add(moveToMake);
 			endMove();
 			return true;
 		}
 		return false;
 	}
 
-	private boolean isValidMoveFromSquare(int row, int column) {
-		if (getSpace(row, column) == null)
-			return false;
-		// This may cause issues when we implement undo/redo if we try invalid
-		// moves before we undo
-		if (getSpace(row, column).getOwner() != Owner.values()[(getPlayerTurn() - 1)] && !isPushPull) {
-			return false;// not your turn
-		}
-		if ((checkStrongerAdjacent(row, column) && !checkFriendlyAdjacent(row, column)) && !isPushPull) {
-			return false;// can't move
+	/**
+	 * 
+	 * @param ownerPiece
+	 *            Coordinate of the owner's piece
+	 * @param opponentPiece
+	 *            Coordinate of the opponent's piece
+	 * @param destination
+	 *            Coordinate of the position that either the opponent's piece
+	 *            will be pushed into or the position the owner's piece will be
+	 *            moved into.
+	 * @return Returns true when a push or pull with the given 3 Coordinate
+	 *         objects would result in a valid move.
+	 */
+
+	public boolean pushOrPull(Coordinate ownerPiece, Coordinate opponentPiece, Coordinate destination) {
+		// TODO: Duplicate code in 3rd nested if-statement. Fix it.
+		// Do you have enough moves?
+		if (this.numMoves >= 2) {
+			// Is the first piece yours, the second theirs, are they
+			// orthonally adjacent, and is your piece higher precedence than theirs?
+			if ((this.currentBoard.getPieceAt(ownerPiece)).getOwner() == this.getOwner()
+					&& (this.currentBoard.getPieceAt(opponentPiece).getOwner() != this.getOwner())
+					&& (ownerPiece.isOrthogonallyAdjacentTo(opponentPiece))
+					&& (this.checkStrongerAdjacent(ownerPiece, opponentPiece))) {
+				// Is the destination next to your piece?
+				if (ownerPiece.isOrthogonallyAdjacentTo(destination)) {
+					RegularMove yourPiece = new RegularMove(this.currentBoard, ownerPiece, destination);
+					RegularMove theirPiece = new RegularMove(this.currentBoard, opponentPiece, ownerPiece);
+					this.currentBoard = yourPiece.execute();
+					this.currentBoard = theirPiece.execute();
+					this.moves.add(yourPiece);
+					this.moves.add(theirPiece);
+					this.numMoves--;
+					endMove();
+					return true;
+					// Or is it next to their piece?
+				} else if (opponentPiece.isOrthogonallyAdjacentTo(destination)) {
+					RegularMove theirPiece = new RegularMove(this.currentBoard, opponentPiece, destination);
+					RegularMove yourPiece = new RegularMove(this.currentBoard, ownerPiece, opponentPiece);
+					this.currentBoard = theirPiece.execute();
+					this.currentBoard = yourPiece.execute();
+					this.moves.add(theirPiece);
+					this.moves.add(yourPiece);
+					this.numMoves--;
+					endMove();
+					return true;
+				} else
+					// Neither, so that isn't a legal move
+					return false;
```

Comment:
stylisticlly is better to have a return like this at the beginning of the method.